scripts/provision_health_checks.py

```diff
@@ -0,0 +1,92 @@
+from itertools import chain
+
+import argparse
+import boto3
+import logging
+import json
+import sys
+
+import more_itertools
+
+from azul import config
+from azul.logging import configure_script_logging
+
+logger = logging.getLogger(__name__)
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description='Dynamically reference and dereference Route53 child health checks '
+                                                 'not managed by Azul.')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--reference', '-R', dest='reference', action='store_true', default=False,
+                       help='Link Azul managed Route53 health check resources to DCP wide composite health check.')
+    group.add_argument('--dereference', '-D', dest='reference', action='store_false', default=False,
+                       help='Unlink Azul managed Route53 health check resources to DCP wide composite health check.')
+    options = parser.parse_args(argv)
+    health_checks = get_health_checks()
+    try:
+        dcp_wide_health_check = health_checks[f'dcp-health-check-{config.deployment_stage}']
```

Comment:
A try/except block seems unnecessary to check if a key exists in a dict. Why not simply:
```suggestion
        if key in health_checks
```

Reply:
I decided to keep it as is, I think calling the dic with the known key has a look up of O(1) and just falling if it is not found contrary to iterating through all the keys in the dic O(n). Let me know if you disagree.

Reply:
Checking membership in a dictionary is O(1). Either way works well, though I've heard it argued "more Pythonic" to do the try-except.